src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -46,8 +47,13 @@
     const shouldDbMigrated = await LocalDatabaseMigration.shouldDbMigrated();
     let scopedMigrationStatus = migrationStatus;
 
+    if (scopedMigrationStatus === MIGRATION_STATUS.MIGRATING) {
+      return null;
+    }
+
     if (isEnteringApp && shouldDbMigrated) {
-      await LocalDatabaseMigration.migrateDb();
+      setMigrationStatus(MIGRATION_STATUS.MIGRATING);
+      if (withMigration) await LocalDatabaseMigration.migrateDb();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here checks if the migration status is 'MIGRATING', and if so, it returns null. Then, if the app is entering and the database should be migrated, it sets the migration status to 'MIGRATING', migrates the database if `withMigration` is true, and finally sets the migration status to 'MIGRATED'. This is a good way to handle the migration process and prevent potential issues.

```diff
+    if (scopedMigrationStatus === MIGRATION_STATUS.MIGRATING) {
+      return null;
+    }
+
+    if (isEnteringApp && shouldDbMigrated) {
+      setMigrationStatus(MIGRATION_STATUS.MIGRATING);
+      if (withMigration) await LocalDatabaseMigration.migrateDb();
+      setMigrationStatus(MIGRATION_STATUS.MIGRATED);
+      scopedMigrationStatus = MIGRATION_STATUS.MIGRATED;
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->